oaknational/Oak-Web-Application
# feat: flag enables posthog tracking of latest URLs in user profiles

## Description
Posthog properties such as Latest Current URL can be useful for implementing feature flags and A/B test overrides in development. However, Posthog does not update the 'Latest' properties on its user profiles by default. Instead they record the user state at the last point that the profile was changed. 

This change updates with an extra property the profile whenever a user navigates to a page using the query parameter `update_tracking_profile=true`. This in turn updates all of the 'Latest' properties with the state of the user.

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2638--oak-web-application.netlify.thenational.academy/pupils/years?update_tracking_profile=true
2. Accept Cookies
3. Open the browser console and find your distinct_id:

![image](https://github.com/user-attachments/assets/5a3fb3ab-22bc-4851-8ff6-2f82a3d48dcd)

![image](https://github.com/user-attachments/assets/7d77c761-b4c8-4c42-8bd0-372f92ea76a3)

4. Go to posthog people https://eu.posthog.com/project/124/persons and find your own profile using the distinct_id
5. Note the Latest Current URL (this should be as per above)
![image](https://github.com/user-attachments/assets/ea0ab23e-d5a7-4dc0-a0b7-198da9a39a85)

6. Navigate to a different OWA url 
7. refresh the posthog profile change and note that the Latest Current URL has NOT updated.
8. In OWA Add the query param `?update_tracking_profile=true` and refresh the page
9. refresh the posthog profile change and note that the Latest Current URL has NOW updated.
 - NB. this can be a little laggy so try refreshing a couple of times
